hw/ip/kmac/doc/_index.md

```diff
@@ -197,33 +197,40 @@ While in an idle state or in the sponge absorbing stage, the value is zero.
 This ensures that the logic does not expose the secret key XORed with the keccak_f results of the prefix to the software.
 In addition to that, the KMAC/SHA3 blocks the software access to the Keccak state when it processes the request from KeyMgr for Key Derivation Function (KDF).
 
-### KeyMgr Interface
+### Application Interface
 
-![](keymgr-intf.svg)
+![](application-interface.svg)
 
-KMAC/SHA3 HWIP has an option to receive the secret key from KeyMgr via sideload key interface.
-The software should set !!CFG.sideload to use the KeyMgr sideloaded key in SW-initiated KMAC operation.
+KMAC/SHA3 HWIP has an option to receive the secret key from the KeyMgr via sideload key interface.
+The software should set !!CFG.sideload to use the KeyMgr sideloaded key for the SW-initiated KMAC operation.
 `keymgr_pkg::hw_key_t` defines the structure of the sideloaded key.
 KeyMgr provides the sideloaded key in two-share masked form regardless of the compile-time parameter `EnMasking`.
 If `EnMasking` is not defined, the KMAC merges the shared key to the unmasked form before uses the key.
 
-KeyMgr may initiate the KMAC operation via the KeyMgr data interface `keymgr_pkg::kmac_data_{req|rsp}_t`.
-KeyMgr sends 64-bit data (`MsgWidth`) in a beat with the message strobe signal.
-The state machine inside the KeyMgr interface logic starts when it receives the first valid data.
-Because this logic sees the first valid data as an initiator, KeyMgr cannot run KDF with an empty message.
-After the logic switches to accept the message bitstream from KeyMgr, it forces the KMAC to use the sideloaded key as a secret key.
+The IP has N number of the application interface. The apps connect to the KMAC IP may initiate the SHA3/cSHAKE/KMAC hashing operation via the application interface `kmac_pkg::app_{req|rsp}_t`.
+The type of the hashing operation is determined in the compile-time parameter `kmac_pkf::AppCfg`.
+In the current version of IP, the IP has three application interfaces, which are KeyMgr, LC_CTRL, and ROM_CTRL.
+KeyMgr uses the KMAC operation with CSR prefix value.
+LC_CTRL and ROM_CTRL use the cSHAKE operation with the compile-time parameter prefixes.
+
+The app sends 64-bit data (`MsgWidth`) in a beat with the message strobe signal.
+The state machine inside the AppIntf logic starts when it receives the first valid data from any of the AppIntf.
+The AppIntf module chooses the winner based on the fixed priority.
+Then it forwards the selected App to the next stage.
+Because this logic sees the first valid data as an initiator, the Apps cannot run the hashing operation with an empty message.
+After the logic switches to accept the message bitstream from the App, it forces the KMAC to use the sideloaded key as a secret key if the hashing operation is the KMAC.
```

Comment:
Suggest (if I understand correctly):
"After the logic switches to accept the message bitstream from the selected App, if the hashing operation is KMAC, the logic forces the sideloaded key to be used as the secret key."

(I'm not sure if this is actually how it works?  Does it decide based on operation == KMAC, or App == KeyMgr)?  The second one makes more sense to me but it's not what you wrote.